### PR TITLE
Fix codecov and test failures

### PR DESCRIPTION
**Pull request by robtaylor**
This *should* fix the issue with linux showing as failing.

### CODE REVIEW

**Comment by robtaylor** [comment]
See https://github.com/doorstop-dev/doorstop/issues/667

**Comment by robtaylor** [comment]
Updated PR to fix two issues causing test and coverage failure:
* codecov required changes to pass the token secret succesfully to the  `execute-tests` workflow
* random changes in plantuml svg generation were seen, so modify publish tests to use a local, known version of plantuml

**Comment by robtaylor** [comment]
Ugh, this changing viewbox parameter in the generated svg is killing me!!

**Comment by robtaylor** [comment]
All working now!

File: .github/workflows/change-coverage.yml [hunk 1]
```diff
@@ -27,6 +27,11 @@ jobs:
           python-version: "3.11"
           architecture: x64
 
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '21'
```
Comment (jacebrowning) [inline .github/workflows/change-coverage.yml:33]: Why is Java required on CI?

File: doorstop/core/tests/test_all.py [hunk 2]
```diff
@@ -575,6 +586,14 @@ def setUp(self):
         self.tree = core.build(cwd=FILES, root=FILES)
         self.document = self.tree.find_document("REQ")
         self.temp = tempfile.mkdtemp()
+        # Use local plantuml of known version
+        plantuml_cmd = "java -Dplantuml.include.path=includes -jar " + os.path.join(
```
Comment (jacebrowning) [inline doorstop/core/tests/test_all.py:590]: I don't think we want to make `java` a dependency of this library. Can you solve this with a Python library?

**Comment by jacebrowning** [comment]
Hi @robtaylor, thanks for getting CI to pass!

Can you open a separate pull request with just the code coverage fixes? As mentioned in the commend above, I don't think we want to add `java` as a dependency in order to get tests to pass. I'd like to review configuration changes and business logic changes separately.

**Comment by robtaylor** [comment]
Hi Jace, yes for sure.

So with respect to plantuml, there isn't really any options - you need to
have java.

That said, as I'm now stripping out the SVG, maybe a better approach here
is to mock calling plantuml. What do you think?

**Comment by robtaylor** [comment]
Hey @jacebrowning, I've put the code changes (plantuml mock) in #669 . This now has just the CI changes.

**Pull request merged**